### PR TITLE
feat(StaticTests): add Response tests and offline currencies

### PR DESCRIPTION
**Pull request by carlosmiei**
- Added static tests to complete our integration tests and test also the parsing done by ccxt in all languages
- We mock the httpResponse and then check if ccxt correctly parsed the result
- test_sync and test_async are now transpiled by the ast-transpiler
- static/data was renamed to static/request
- static/response contains these new tests
- the cli command to generate the report has changed as well:
```
//  create a request report  (--response flag)
node cli.js n binance fetchOpenOrders "LTC/USDT" undefined 1 --request

// create a response report (--response flag)

node cli.js n binance fetchOpenOrders "LTC/USDT" undefined 1 --response   
```
- These tests are very similar to the request tests and the process is the same. We need to generate a report and add it in the correct file eg: (static/response/binance.json)

How to run all static tests (request + response): `npm run static-tests`

### CODE REVIEW

File: php/async/Exchange.php [hunk 1]
```diff
@@ -79,8 +79,16 @@ public function set_request_browser($connector_options = array()) {
...
-        return React\Async\async(function () use ($url, $method, $headers, $body) {
+        $fetch_result = $this->fetch_result;
+        return React\Async\async(function () use ($url, $method, $headers, $body, $fetch_result) {
 
+            if ($fetch_result !== null) {
+                // used in tests
+                // it's not the prettiest solution but
+                // couldn't find a better way to do it
+                // feel free to suggest a better alternative
+                return $fetch_result;
```
Comment (carlosmiei) [inline php/async/Exchange.php:90]: @ttodua I tried a lot of things (dynamic overrides, dynamic class extends, etc) but couldn't find a better alternative in PHP to mock the https response. do you have a better suggestion?

**Comment by carlosmiei** [comment]
@ttodua also, I'm aware that the ast-transpiler is not transpiling some objects with arrays perfectly, but this is not a syntax issue, and I will try to address it soon. 
image

**Comment by ttodua** [comment]
> @ttodua also, I'm aware that the ast-transpiler is not transpiling some objects with arrays perfectly, but this is not a syntax issue, and I will try to address it soon. image

so it's just formatting imperfection right? if so, not a much problem at all

File: php/async/Exchange.php (hunk 1 above)
Comment (ttodua) [reply php/async/Exchange.php:90]: i'll check some things and lyk

**Pull request merged**